esrally/chart_generator.py

```diff
@@ -1564,7 +1566,12 @@ def load_race_configs(cfg, chart_type, chart_spec_path=None):
     def add_configs(race_configs_per_lic, flavor_name="oss", lic="oss"):
         configs_per_lic = []
         for race_config in race_configs_per_lic:
-            configs_per_lic.append(RaceConfig(track=t, cfg=race_config, flavor=flavor_name, es_license=lic))
+            configs_per_lic.append(
+                RaceConfig(track=load_track(cfg, item["track"], race_config.get("track-params", {})),
```

Comment:
I think you need to do the same when loading a single track here: https://github.com/elastic/rally/blob/ae3e14ac40548029c5fc8ed54a29c6b3a7c5ddbb/esrally/chart_generator.py#L1609

(assuming users can pass `--track-params` when invoking the chart generator which should be supported in that case)

Also we're loading the very same track repeatedly (which is quite time-consuming already) and I wonder whether we should avoid this and use a cached version if a race configuration doesn't use track parameters (i.e. the dictionary is empty). This would speed up the standard case significantly IMHO.

Reply:
I address both comments in https://github.com/elastic/rally/pull/686/commits/4a93a9f7053422950d44f1a4929a0891a8cbdc26, could you take another pass please?